src/fstab-generator/fstab-generator.c

```diff
@@ -358,6 +358,15 @@ static int add_mount(
                 "Documentation=man:fstab(5) man:systemd-fstab-generator(8)\n",
                 source);
 
+        if (strneq(fstype, "nfs", 3) && !automount &&
+            fstab_test_yes_no_option(opts, "bg\0" "fg\0")) {
+                opt_bg = strjoin("x-systemd.mount-timeout=infinity,retry=10000,", opts, ",fg");
```

Comment:
I think this really deserves a comment here, that explains why "10000" is used here (i.e. there should be a reference to the fact that this is what is used by mount.nfs' own "bg" logic...

(I am not sure I like setting the mount timeout like this here too much... i.e. formatting this as a string first, and then parsing this out a few lines later isn't precisely pretty... but ok)